File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/NutrientLevelListAdapter.java
package openfoodfacts.github.scrachx.openfood.views.adapters;

import android.content.Context;
import android.support.v7.content.res.AppCompatResources;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

import openfoodfacts.github.scrachx.openfood.R;
import openfoodfacts.github.scrachx.openfood.models.NutrientLevelItem;

import static openfoodfacts.github.scrachx.openfood.utils.Utils.bold;

public class NutrientLevelListAdapter extends RecyclerView.Adapter<NutrientLevelListAdapter.NutrientViewHolder> {

    private Context context;
    private List<NutrientLevelItem> nutrientLevelItems;

    public NutrientLevelListAdapter(Context context, List<NutrientLevelItem> navDrawerItems) {
        this.context = context;
        this.nutrientLevelItems = navDrawerItems;
    }

    @Override
    public NutrientViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.nutrient_lvl_list_item, parent, false);
        return new NutrientViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(NutrientViewHolder holder, int position) {

        NutrientLevelItem nutrientLevelItem = nutrientLevelItems.get(position);


        if (nutrientLevelItem.getIcon() <= 0) {
            holder.imgIcon.setVisibility(View.GONE);
        } else {
            holder.imgIcon.setImageDrawable(AppCompatResources.getDrawable(context, nutrientLevelItem.getIcon()));
            holder.imgIcon.setVisibility(View.VISIBLE);
        }


        TextView titleTxt = holder.txtTitle;
        titleTxt.setText("");
        titleTxt.append(nutrientLevelItem.getValue());
        titleTxt.append(" ");
        titleTxt.append(bold(nutrientLevelItem.getCategory()));
        titleTxt.append("\n");
        titleTxt.append(nutrientLevelItem.getLabel());

    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        return nutrientLevelItems.size();
    }


    class NutrientViewHolder extends RecyclerView.ViewHolder {
        ImageView imgIcon;
        TextView txtTitle;

        public NutrientViewHolder(View itemView) {
            super(itemView);
            imgIcon = itemView.findViewById(R.id.imgLevel);
            txtTitle = itemView.findViewById(R.id.descriptionLevel);


        }
    }

}

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/shared/adapters/NutrientLevelListAdapter.kt
package openfoodfacts.github.scrachx.openfood.features.shared.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.View.NO_ID
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.content.res.AppCompatResources
import androidx.core.text.bold
import androidx.core.text.buildSpannedString
import androidx.recyclerview.widget.RecyclerView
import openfoodfacts.github.scrachx.openfood.R
import openfoodfacts.github.scrachx.openfood.features.shared.adapters.NutrientLevelListAdapter.NutrientViewHolder
import openfoodfacts.github.scrachx.openfood.models.NutrientLevelItem

class NutrientLevelListAdapter(
    private val context: Context,
    private val nutrientLevelItems: List<NutrientLevelItem>
) : RecyclerView.Adapter<NutrientViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
        NutrientViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.nutrient_lvl_list_item, parent, false)
        )

    override fun onBindViewHolder(holder: NutrientViewHolder, position: Int) {
        val (category, value, label, icon) = nutrientLevelItems[position]

        if (icon == NO_ID) {
            holder.imgIcon.visibility = View.GONE
        } else {
            holder.imgIcon.setImageDrawable(AppCompatResources.getDrawable(context, icon))
            holder.imgIcon.visibility = View.VISIBLE
        }

        holder.txtTitle.text = buildSpannedString {
            append(value)
            append(" ")
            bold { append(category) }
            append("\n")
            append(label)
        }
    }

    override fun getItemId(position: Int) = position.toLong()

    override fun getItemCount() = nutrientLevelItems.size

    class NutrientViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val imgIcon: ImageView = itemView.findViewById(R.id.imgLevel)
        val txtTitle: TextView = itemView.findViewById(R.id.descriptionLevel)
    }
}